applications/ContactStructuralMechanicsApplication/custom_utilities/interface_preprocess.cpp

```diff
@@ -68,32 +64,29 @@ void InterfacePreprocessCondition::GenerateInterfacePart(
         std::unordered_map<IndexType, Properties::Pointer> new_properties;
         if (contact_property_id == 0) new_properties = CreateNewProperties();
 
-        if (dimension == 2) {
-            // We iterate over the elements and check the nodes on the interface
-            for (auto it_elem = mrMainModelPart.ElementsBegin(); it_elem != mrMainModelPart.ElementsEnd(); ++it_elem) {
-                GeometryType& this_geometry = it_elem->GetGeometry();
-                Properties::Pointer p_prop = (contact_property_id == 0) ? new_properties[it_elem->pGetProperties()->Id()] : mrMainModelPart.CreateNewProperties(contact_property_id);
-                KRATOS_DEBUG_ERROR_IF(p_prop == nullptr) << "ERROR:: Property not well initialized" << std::endl;
-
-                if (this_geometry.LocalSpaceDimension() == 2) {
-                    for (IndexType i_edge = 0; i_edge < this_geometry.EdgesNumber(); ++i_edge)
-                        GenerateEdgeCondition(rInterfacePart, p_prop, this_geometry.Edges()[i_edge], simplest_geometry, cond_counter, cond_id);
-                } else {
-                    GenerateEdgeCondition(rInterfacePart, p_prop, this_geometry, simplest_geometry, cond_counter, cond_id);
+        // We iterate over the elements and check the nodes on the interface
+        for (auto& r_elem : mrMainModelPart.Elements()) {
+            GeometryType& r_geometry = r_elem.GetGeometry();
+            const std::size_t working_space_dimension = r_geometry.WorkingSpaceDimension();
+            const std::size_t local_space_dimension = r_geometry.LocalSpaceDimension();
+
+            Properties::Pointer p_prop = (contact_property_id == 0) ? new_properties[r_elem.pGetProperties()->Id()] : mrMainModelPart.CreateNewProperties(contact_property_id);
+            KRATOS_DEBUG_ERROR_IF(p_prop == nullptr) << "ERROR:: Property not well initialized" << std::endl;
```

Comment:
```suggestion
            KRATOS_DEBUG_ERROR_IF_NOT(p_prop) << "ERROR:: Property not well initialized" << std::endl;
```

Reply:
I am not sure about the change..., and the checks already finished

Reply:
You can check pointers with true/false if they are valid
Just minor

Reply:
I see, but I preferred to compile first, late